src/Network.cpp

```diff
@@ -406,10 +406,13 @@ Network::Netresult Network::get_scored_moves_internal(
     assert(rotation >= 0 && rotation <= 7);
     constexpr int width = 19;
     constexpr int height = 19;
-    assert(INPUT_CHANNELS == planes.size());
-    std::vector<net_t> input_data;
-    auto convolve_channels = conv_pol_w.size() / conv_pol_b.size();
-    std::vector<net_t> output_data(convolve_channels * width * height);
+    constexpr int input_channels = INPUT_CHANNELS;
+    assert(input_channels == planes.size());
+    const auto input_size = input_channels * width * height;
+    const auto output_channels = int(conv_pol_w.size() / conv_pol_b.size());
+    const auto output_size = output_channels * width * height;
+    std::vector<net_t> input_data(input_size);
```

Comment:
This undoes part of the optimization! It forces the array to be default-initialized even though we will overwrite it instantly. 

The previous code doesn't initialize the array, but reserves enough place ahead of time (so there is only a single allocation), and then emplace_backs the elements into it, which is equivalent to the pointer code, but (IMHO) better C++ style.

Reply:
Yep, you're totally correct about reserve / fill constructor. 

I think that emplace_back and push_back get optimized to the same code but I'll check with assembler / profiler at some later date.